### ISSUE TITLE
Implement cost regularization

### ISSUE THREAD

**Issue by sarambulo** [nikitachaudharicodes/ProjeXion#6]
Create a `cost_regularization.py` module inside the `models` directory that implements the class `CostRegularizer`. This class should inherit from `torch.nn.Module` and should take as input a HxWxDx32 tensor and generate an output tensor of size HxWxDx1 using a 3D-Unet:

*To further lessen the computational requirement, we reduce the 32-channel cost volume to 8-channel after the first 3D convolutional layer, and change the convolutions within each scale from 3 layers to 2 layers. The last convolutional layer outputs a 1-channel volume. We finally apply the softmax operation along the depth direction for probability normalization.*

![Image](https://github.com/user-attachments/assets/b1e14a1c-7e8a-4c9a-aecf-dc16b2aca3c3)
